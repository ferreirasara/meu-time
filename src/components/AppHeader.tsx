import { BarChartOutlined, InfoOutlined, LogoutOutlined } from "@ant-design/icons"
import { Button, Space, Typography, theme } from "antd"
import { Header } from "antd/lib/layout/layout"
import { useState } from "react";
import { InfoModal } from "./InfoModal";

export const AppHeader = () => {
  const [infoModalOpen, setInfoModalOpen] = useState<boolean>(false);
  const {
    token: { colorBgContainer },
  } = theme.useToken();

  const handleLogout = () => {
    localStorage.removeItem("api-key");
    localStorage.removeItem("user-data");
    localStorage.removeItem("requests-data");
    window.location.reload();
  }

  return <Header
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between'
    }}
  >
    <Typography.Title
      level={2}
      style={{ margin: 0, color: colorBgContainer }}
    >
      <BarChartOutlined /> Meu Time
    </Typography.Title>
    <Space>
      <Button
        ghost
        icon={<InfoOutlined />}
        onClick={() => setInfoModalOpen(true)}
      >
        Informações
      </Button>
      <Button
        ghost
        icon={<LogoutOutlined />}
        onClick={handleLogout}
      >
        Sair
      </Button>
    </Space>
    <InfoModal
      infoModalOpen={infoModalOpen}
      onClose={() => setInfoModalOpen(false)}
    />
  </Header>
}