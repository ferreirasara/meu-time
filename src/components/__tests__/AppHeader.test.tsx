import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { AppHeader } from '../AppHeader';

describe('AppHeader', () => {
  it('match snapshot', () => {
    const appHeader = renderer.create(<AppHeader />).toJSON();
    expect(appHeader).toMatchSnapshot();
  });

  it('renders title text', () => {
    const { getByText } = render(<AppHeader />);

    const titleEl = getByText(/Meu Time/);
    expect(titleEl).toBeInTheDocument();
  });

  it('renders info button', () => {
    const { getByText } = render(<AppHeader />);

    const infoButton = getByText(/Informações/);
    expect(infoButton).toBeInTheDocument();
  });

  it('renders logout button', () => {
    const { getByText } = render(<AppHeader />);

    const buttonEl = getByText(/Sair/);
    expect(buttonEl).toBeInTheDocument();
  });
})