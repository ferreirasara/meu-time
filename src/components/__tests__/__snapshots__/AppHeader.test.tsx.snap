// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader match snapshot 1`] = `
<header
  className="ant-layout-header css-dev-only-do-not-override-15rg2km"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <h2
    className="ant-typography css-dev-only-do-not-override-15rg2km"
    style={
      Object {
        "WebkitLineClamp": undefined,
        "color": "#ffffff",
        "margin": 0,
      }
    }
  >
    <span
      aria-label="bar-chart"
      className="anticon anticon-bar-chart"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="bar-chart"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-600-80h56c4.4 0 8-3.6 8-8V560c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V384c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v320c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V462c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v242c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v400c0 4.4 3.6 8 8 8z"
        />
      </svg>
    </span>
     Meu Time
  </h2>
  <div
    className="ant-space css-dev-only-do-not-override-15rg2km ant-space-horizontal ant-space-align-center"
    style={Object {}}
  >
    <div
      className="ant-space-item"
      style={
        Object {
          "marginRight": 8,
        }
      }
    >
      <button
        className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-background-ghost"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="ant-btn-icon"
        >
          <span
            aria-label="info"
            className="anticon anticon-info"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="info"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M448 224a64 64 0 10128 0 64 64 0 10-128 0zm96 168h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8h64c4.4 0 8-3.6 8-8V400c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
        </span>
        <span>
          Informações
        </span>
      </button>
    </div>
    <div
      className="ant-space-item"
      style={Object {}}
    >
      <button
        className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-default ant-btn-background-ghost"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="ant-btn-icon"
        >
          <span
            aria-label="logout"
            className="anticon anticon-logout"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="logout"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
              />
            </svg>
          </span>
        </span>
        <span>
          Sair
        </span>
      </button>
    </div>
  </div>
</header>
`;
